lnrpc/walletrpc/walletkit.proto

```diff
@@ -397,13 +404,54 @@ message Account {
     // Whether the wallet stores private keys for the account.
     bool watch_only = 8;
 }
+
+message AddressProperty {
+    // The address encoded using a bech32 format.
+    string address = 1;
+
+    // Denotes if the address is a change address.
+    bool is_internal = 2;
+
+    // The balance of an address.
+    int64 balance = 3;
+}
+
+message AccountWithAddresses {
+    // The name used to identify the account.
+    string name = 1;
+
+    /*
+    The type of addresses the account supports.
+    AddressType                       | External Branch | Internal Branch
+    ---------------------------------------------------------------------
+    WITNESS_PUBKEY_HASH               | P2WPKH          | P2WPKH
+    NESTED_WITNESS_PUBKEY_HASH        | NP2WPKH         | NP2WPKH
+    HYBRID_NESTED_WITNESS_PUBKEY_HASH | NP2WPKH         | P2WPKH
+    */
+    AddressType address_type = 2;
+
+    /*
+    The derivation path corresponding to the account public key. This will
+    always be empty for the default imported account in which single public keys
+    are imported into.
+    */
+    string derivation_path = 3;
+
+    /*
+    List of address, its type internal/external & balance.
+    Note that the order of addresses will be random.
```

Comment:
This probably needs some additional context. How about "Note that the order of addresses will be random and not according to the derivation index, since that information is not stored by the underlying wallet."